gitops/pac/.tekton/argo-registration.yaml

```diff
@@ -4,7 +4,7 @@ metadata:
   name: registration-pipeline
   annotations:
     pipelinesascode.tekton.dev/on-target-branch: "[main]"
-    pipelinesascode.tekton.dev/on-event: "[pull_request]"
+    pipelinesascode.tekton.dev/on-event: "[push]"
```

Comment:
I was under the impression that Pull is the right one. Because if it is push, this means that the customer will have to merge their kubeconfig into Github which includes private info like token, cluster names which is not recommended right?

Reply:
not into GitHub... into their private repository, which may be GitHub but also Bitbucket or GitLab. This is until we have integrated with something like Vault.
The difference between pull_request and push is really the git event in the process. When you create a pull_request, the "pull_request" event is triggered. When the pull request is merged the "push" event is triggered (because that's what it is)

Reply:
[Here](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) it is described what triggers a pull request event.
It seems that it would also be possible to use [type "close"](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-when-a-pull-request-merges) with GitHub but I don't know whether it would work with other git repositories.

Reply:
Yes, it could be any of the Git providers.
I think if we are going to use Vault (future) and maintain kubeconfigs on the repo, then better to use a normal Push than pull request on close.
But do you think we could do push for specific files as described [here](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-only-when-a-push-affects-specific-files)?

Reply:
I haven't seen a way to do the filtering in Pipelines as Code but it can surely be done in the PipelineRun within a git task